classes/ActionScheduler_wpPostStore.php

```diff
@@ -11,28 +11,137 @@ class ActionScheduler_wpPostStore extends ActionScheduler_Store {
 	/** @var DateTimeZone */
 	protected $local_timezone = NULL;
 
-	public function save_action( ActionScheduler_Action $action, DateTime $date = NULL ){
+	/**
+	 * Mappings for Action fields to post fields.
+	 *
+	 * @var array
+	 */
+	protected $field_map = array(
+		'action_id'            => 'ID',
+		'hook'                 => 'post_title',
+		'status'               => 'post_status',
+		'scheduled_date_gmt'   => 'post_date_gmt',
+		'scheduled_date_local' => 'post_date',
+		'args'                 => 'post_content',
+		'last_attempt_gmt'     => 'post_modified_gmt',
+		'last_attempt_local'   => 'post_modified',
+		'claim_id'             => 'post_password',
```

Comment:
I'm not really sure we want claims to be updated via a generic `update_action()` call. There is a fairly specific process already for staking a claim, and allowing it to be set outside that could easily lead to unexpected results (like invalid claim IDs, or worse, reused claim IDs).

Reply:
I agree. This makes me think all the more that it would be good to have the generic update being protected/private, and have specific dedicated methods for the various pieces of data.